src/lv_message_efficient.cc

```diff
@@ -173,24 +208,44 @@ namespace grpc_labview
         }
         else
         {
-            LVMessageEfficient nestedMessage(metadata);
-            const char* lv_ptr = (this->GetLVClusterHandleSharedPtr()) + fieldInfo.clusterOffset;
-            nestedMessage.SetLVClusterHandle(lv_ptr);
+            auto nestedClusterPtr = _LVClusterHandle + fieldInfo.clusterOffset;
+            LVMessageEfficient nestedMessage(metadata, nestedClusterPtr);
             protobuf_ptr = ctx->ParseMessage(&nestedMessage, protobuf_ptr);
         }
         return protobuf_ptr;
     }
 
+    //---------------------------------------------------------------------
+    //---------------------------------------------------------------------
     void LVMessageEfficient::PostInteralParseAction()
     {
+        if (_oneofContainerToSelectedIndexMap.size() > 0)
+        {
+            // Must iterate over _elements and not _mappedElements since all oneof selected_index fields use -1 for the field number
+            // and there can be multiple oneof fields in a message.
+            for (auto& fieldMetadata : _metadata->_elements)
+            {
+                if (fieldMetadata->isInOneof && fieldMetadata->protobufIndex < 0)
+                {
+                    // This field is the selected_index field of a oneof. This field only exists in the cluster
+                    // and is not a field in the message.
+                    if (_oneofContainerToSelectedIndexMap.find(fieldMetadata->oneofContainerName) != _oneofContainerToSelectedIndexMap.end())
```

Comment:
This does an extra map lookup with operator[] below. Save the iterator and use it in the assignment instead.

FYI, there's similar code in CopyToCluster, maybe it could be extracted into an LVMessage method.